common/flatpak-dir.c

```diff
@@ -2818,6 +2829,9 @@ flatpak_dir_load_deployed (FlatpakDir        *self,
   FlatpakDeploy *deploy;
   gsize metadata_size;
 
+  if (!flatpak_dir_ensure_repo (self, cancellable, error))
```

Comment:
I think we should check this after the flatpak_dir_get_if_deployed() call, because that will return a more useful error in the no-repo case (i.e. the ref is not installed, rather than some internalish error).

Reply:
Makes sense. I think then the only situation where `flatpak_dir_ensure_repo` could fail here is programmer error, but we can leave it.